src/backend/joanie/core/api/admin/__init__.py

```diff
@@ -232,6 +237,71 @@ def get_serializer_context(self):
         context["organization_id"] = self.kwargs["organization_id"]
         return context
 
+    def create(self, request, *args, **kwargs):
+        """
+        Create organization access and update the organization signatories
+        for signature procedures that are currently ongoing.
+        """
+        response = super().create(request, *args, **kwargs)
+
+        if self.request.data.get("role") == enums.OWNER:
+            organization = models.Organization.objects.get(
+                pk=self.request.parser_context.get("kwargs").get("organization_id")
+            )
+            update_organization_signatories_contracts_task.delay(
+                organization_id=organization.id
+            )
+
+        return response
```

Comment:
Instead of overriding `create` you should only focus on `perform_create` allowing you to retrieve the saved instance and your will be simple as that:

```suggestion
    def perform_create(self, serializer):
        instance = serializer.save()

        if instance.role == enums.OWNER:
            update_organization_signatories_contracts_task.delay(
                organization_id=instance.organization.id
            )
```

Reply:
Indeed, it is much simpler this way. Thanks :pray: